shell/browser/api/electron_api_cookies.cc

```diff
@@ -171,25 +171,95 @@ base::Time ParseTimeProperty(const std::optional<double>& value) {
   return base::Time::FromSecondsSinceUnixEpoch(*value);
 }
 
-std::string_view InclusionStatusToString(net::CookieInclusionStatus status) {
-  if (status.HasExclusionReason(net::CookieInclusionStatus::EXCLUDE_HTTP_ONLY))
-    return "Failed to create httponly cookie";
-  if (status.HasExclusionReason(
-          net::CookieInclusionStatus::EXCLUDE_SECURE_ONLY))
-    return "Cannot create a secure cookie from an insecure URL";
-  if (status.HasExclusionReason(
-          net::CookieInclusionStatus::EXCLUDE_FAILURE_TO_STORE))
-    return "Failed to parse cookie";
-  if (status.HasExclusionReason(
-          net::CookieInclusionStatus::EXCLUDE_INVALID_DOMAIN))
-    return "Failed to set cookie with an invalid domain attribute";
-  if (status.HasExclusionReason(
-          net::CookieInclusionStatus::EXCLUDE_INVALID_PREFIX))
-    return "Failed because the cookie violated prefix rules.";
-  if (status.HasExclusionReason(
-          net::CookieInclusionStatus::EXCLUDE_NONCOOKIEABLE_SCHEME))
-    return "Cannot set cookie for current scheme";
-  return "Setting cookie failed";
+const std::string InclusionStatusToString(net::CookieInclusionStatus status) {
+  // See net/cookies/cookie_inclusion_status.h for cookie error descriptions.
+  constexpr std::pair<net::CookieInclusionStatus::ExclusionReason, const char*>
+      exclusion_reasons[] = {
```

Comment:
This should be a `constexpr std::array<std::pair<net::CookieInclusionStatus::ExclusionReason, std::string_view>, net::CookieInclusionStatus::ExclusionReason::NUM_EXCLUSION_REASONS>`:

- string length will unambiguously be preserved for the `std::string +=` operator
- Will replace the `static_assert` and is a better alternative, since it will break any time any item is added, not just when `NUM_EXCLUSION_REASONS` grows to 33 (current value is 27).